tests/e2e/inference_pool_test.go

```diff
@@ -30,24 +30,34 @@ func TestInferencePoolIntegration(t *testing.T) {
 	egSelector := "gateway.envoyproxy.io/owning-gateway-name=inference-pool-with-aigwroute"
 	requireWaitForGatewayPodReady(t, egSelector)
 
-	// Test connectivity to inferencePool + inference pods with valid metrics.
-	t.Run("endpointpicker_with_aigwroute_valid_pod_metrics", func(t *testing.T) {
-		testInferenceGatewayConnectivityByModel(t, egSelector, "meta-llama/Llama-3.1-8B-Instruct")
+	// Test connectivity to inferencePool + header match + inference pods with valid metrics, should return 200.
+	t.Run("endpointpicker_with_aigwroute_matched_header", func(t *testing.T) {
+		testInferenceGatewayConnectivityByModel(t, egSelector, "meta-llama/Llama-3.1-8B-Instruct", map[string]string{"Authorization": "sk-abcdefghijklmnopqrstuvwxyz"}, http.StatusOK)
+	})
+
+	// Test connectivity to inferencePool + header match + inference pods with valid metrics, should return 200.
+	t.Run("endpointpicker_with_aigwroute_matched_header", func(t *testing.T) {
+		testInferenceGatewayConnectivityByModel(t, egSelector, "meta-llama/Llama-3.1-8B-Instruct", map[string]string{"Authorization": "sk-zyxwvutsrqponmlkjihgfedcba"}, http.StatusOK)
 	})
 
-	// Test connectivity to inferencePool + inference pods with invalid metrics.
+	// Test connectivity to inferencePool + unmatched route + inference pods with valid metrics, should return 404 directly.
+	t.Run("endpointpicker_with_aigwroute_unmatched", func(t *testing.T) {
+		testInferenceGatewayConnectivityByModel(t, egSelector, "meta-llama/Llama-3.1-8B-Instruct", map[string]string{}, http.StatusNotFound)
```

Comment:
Same

Reply:
I am off computer now, could polish it later